iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/ExecutableNotOwnedByRootCheck.java

```diff
@@ -0,0 +1,127 @@
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.common.api.checks.SecondaryLocation;
+import org.sonar.iac.common.api.tree.HasTextRange;
+import org.sonar.iac.docker.checks.utils.Chmod;
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+import org.sonar.iac.docker.tree.api.Argument;
+import org.sonar.iac.docker.tree.api.Flag;
+import org.sonar.iac.docker.tree.api.TransferInstruction;
+
+@Rule(key = "S6504")
+public class ExecutableNotOwnedByRootCheck implements IacCheck {
+
+  private static final String MESSAGE = "Make sure no write permissions are assigned to the executable.";
+  private static final String MESSAGE_SECONDARY_OTHER_EXEC = "Other executable file.";
+  private static final String MESSAGE_SECONDARY_CHOWN = "Sensitive file owner.";
+  private static final Set<String> SENSITIVE_FILE_EXTENSION = Set.of("exe", "py", "rb", "pl", "lua", "js", "lisp", "sh", "jar", "war",
+    "run", "bin", "bat", "ps1");
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(TransferInstruction.class, ExecutableNotOwnedByRootCheck::checkTransferInstruction);
+  }
+
+  private static void checkTransferInstruction(CheckContext ctx, TransferInstruction transferInstruction) {
+    Flag sensitiveChownFlag = getSensitiveChownFlag(transferInstruction);
+    if (sensitiveChownFlag != null) {
+      List<Argument> sensitiveFiles = getSensitiveFiles(transferInstruction.srcs());
+      Chmod chmod = getChmod(transferInstruction);
+
+      if (chmod == null) {
+        if (!sensitiveFiles.isEmpty()) {
+          reportIssue(ctx, sensitiveChownFlag, sensitiveFiles);
+        }
+      } else if (isChmodWriteSensitive(chmod) && (isChmodExecuteSensitive(chmod) || !sensitiveFiles.isEmpty())) {
+        reportIssue(ctx, sensitiveChownFlag, sensitiveFiles);
+      }
+    }
+  }
+
+  private static void reportIssue(CheckContext ctx, Flag sensitiveChownFlag, List<Argument> sensitiveFiles) {
+    if (sensitiveFiles.isEmpty()) {
+      ctx.reportIssue(sensitiveChownFlag, MESSAGE);
+    } else {
+      HasTextRange primaryLocation = sensitiveFiles.get(0);
+      List<SecondaryLocation> secondaryLocations = new ArrayList<>();
+      for (Argument otherExecutable : sensitiveFiles.subList(1, sensitiveFiles.size())) {
+        secondaryLocations.add(new SecondaryLocation(otherExecutable, MESSAGE_SECONDARY_OTHER_EXEC));
+      }
+      secondaryLocations.add(new SecondaryLocation(sensitiveChownFlag, MESSAGE_SECONDARY_CHOWN));
+      ctx.reportIssue(primaryLocation, MESSAGE, secondaryLocations);
+    }
+  }
+
+  @CheckForNull
+  private static Flag getSensitiveChownFlag(TransferInstruction transferInstruction) {
+    return transferInstruction.options().stream()
+      .filter(f -> f.name().equals("chown"))
+      .filter(ExecutableNotOwnedByRootCheck::isSensitiveUser)
+      .findFirst().orElse(null);
+  }
+
+  private static boolean isSensitiveUser(Flag chownFlag) {
```

Comment:
There are different method names patterns: `isSensitiveUser` vs. `isFileSensitive`, `isChmodWriteSensitive`, etc.
It will be nice to have one common convention. e.g. `Sensitive` is always at the end of the name